riptide-stream/src/test/java/org/zalando/riptide/stream/StreamsTest.java

```diff
@@ -0,0 +1,296 @@
+import static org.zalando.riptide.stream.Streams.forEach;
+import static org.zalando.riptide.stream.Streams.streamOf;
+import static org.zalando.riptide.Route.listOf;
+
+public class StreamsTest {
+
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    private final String baseUrl = "https://api.example.com";
+    private final URI url = URI.create(baseUrl + "/accounts");
+
+    private final Rest unit;
+    private final MockRestServiceServer server;
+
+    public StreamsTest() {
+        final ObjectMapper mapper = new ObjectMapper().findAndRegisterModules();
+        MockSetup setup = new MockSetup(baseUrl, Arrays.asList(Streams.streamConverter(mapper)));
+        this.server = setup.getServer();
+        this.unit = setup.getRest();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldCallConsumerWithList() throws Exception {
+        final InputStream stream = new ClassPathResource("account-list.json").getInputStream();
+    
+        server.expect(requestTo(url)).andRespond(
+                withSuccess()
+                        .body(new InputStreamResource(stream))
+                        .contentType(APPLICATION_JSON));
+    
+        final ThrowingConsumer<List<AccountBody>> verifier = mock(ThrowingConsumer.class);
+    
+        unit.get("/accounts").dispatch(status(),
+                on(OK).call(streamOf(listOf(AccountBody.class)), forEach(verifier)),
+                anyStatus().call(this::fail)).get();
+    
+        verify(verifier).accept(Arrays.asList(
+                new AccountBody("1234567890", "Acme Corporation"),
+                new AccountBody("1234567891", "Acme Company"),
+                new AccountBody("1234567892", "Acme GmbH"),
+                new AccountBody("1234567893", "Acme SE")));
+        verify(verifier, times(1)).accept(any(List.class));
+    }
+
+    @Test
+    public void shouldCallConsumerWithArray() throws Exception {
+        final InputStream stream = new ClassPathResource("account-list.json").getInputStream();
+    
+        server.expect(requestTo(url)).andRespond(
+                withSuccess()
+                        .body(new InputStreamResource(stream))
+                        .contentType(APPLICATION_JSON));
+    
+        @SuppressWarnings("unchecked")
+        final ThrowingConsumer<AccountBody[]> verifier = mock(ThrowingConsumer.class);
+    
+        unit.get("/accounts").dispatch(status(),
+                on(OK).call(streamOf(AccountBody[].class), forEach(verifier)),
+                anyStatus().call(this::fail)).get();
+    
+        verify(verifier).accept(new AccountBody[] {
+                new AccountBody("1234567890", "Acme Corporation"),
+                new AccountBody("1234567891", "Acme Company"),
+                new AccountBody("1234567892", "Acme GmbH"),
+                new AccountBody("1234567893", "Acme SE") });
+        verify(verifier, times(1)).accept(any(AccountBody[].class));
```

Comment:
Since Java 8 the any method doesn't need to get a parameter, type inference is now powerful enough.